packages/core/admin/admin/src/content-manager/pages/ListView/reducer.js

```diff
@@ -62,31 +62,46 @@ const listViewReducer = (state = initialState, action) =>
             key: `__${name}_key__`,
           };
 
-          if (attributes[name].type === 'component') {
-            const componentName = attributes[name].component;
-            const mainField = get(
-              state,
-              ['components', componentName, 'settings', 'mainField'],
-              null
-            );
+          switch (attributes[name].type) {
+            case 'component': {
+              const componentName = attributes[name].component;
+              const mainFieldName = get(
+                state,
+                ['components', componentName, 'settings', 'mainField'],
+                null
+              );
+              const mainFieldAttributes = get(state, [
```

Comment:
This should be singular as we retrieve only one attribute.

Reply:
No, we retrieve the whole attributes object, which always contains `type`, but sometimes also `default`. I think plural makes sense here.

Reply:
Well it is an attribute definition which always has a `type` and can have validations such as `required,min,max` and also `default` but it is still an attribute definition of the content type.